frontend/src/components/pages/teacher/ClassroomsPage.tsx

```diff
@@ -32,88 +37,14 @@ const ClassroomsPage = (): React.ReactElement => {
 
   const [isModalOpen, setIsModalOpen] = useState(false);
 
-  const classrooms = [
-    {
-      id: "1",
-      name: "David",
-      studentCount: 100,
-      assessmentCount: 100,
-      grade: Grade.GRADE_4,
-      activeAssessments: 100,
-    },
-    {
-      id: "2",
-      name: "David 2",
-      studentCount: 200,
-      assessmentCount: 200,
-      grade: Grade.GRADE_2,
-      activeAssessments: 200,
-    },
-    {
-      id: "3",
-      name: "David 3",
-      studentCount: 200,
-      assessmentCount: 200,
-      grade: Grade.GRADE_2,
-      activeAssessments: 200,
-    },
-    {
-      id: "4",
-      name: "David 4",
-      studentCount: 200,
-      assessmentCount: 200,
-      grade: Grade.GRADE_2,
-      activeAssessments: 200,
-    },
-    {
-      id: "5",
-      name: "David 5",
-      studentCount: 200,
-      assessmentCount: 200,
-      grade: Grade.GRADE_2,
-      activeAssessments: 200,
-    },
-    {
-      id: "6",
-      name: "David 6",
-      studentCount: 200,
-      assessmentCount: 200,
-      grade: Grade.GRADE_2,
-      activeAssessments: 200,
-    },
-    {
-      id: "7",
-      name: "David 7",
-      studentCount: 200,
-      assessmentCount: 200,
-      grade: Grade.GRADE_2,
-      activeAssessments: 200,
-    },
-    {
-      id: "8",
-      name: "David 8",
-      studentCount: 200,
-      assessmentCount: 200,
-      grade: Grade.GRADE_2,
-      activeAssessments: 200,
-    },
-    {
-      id: "9",
-      name: "David 9",
-      studentCount: 200,
-      assessmentCount: 200,
-      grade: Grade.GRADE_2,
-      activeAssessments: 200,
-    },
-    {
-      id: "10",
-      name: "David 10",
-      studentCount: 200,
-      assessmentCount: 200,
-      grade: Grade.GRADE_2,
-      activeAssessments: 200,
-    },
-  ];
+  const { authenticatedUser } = useContext(AuthContext);
+
+  const { loading, error, data } = useQuery(GET_CLASSES_BY_TEACHER, {
+    fetchPolicy: "cache-and-network",
+    variables: { teacherId: authenticatedUser?.id },
```

Comment:
nit: my preference would be something like this:
```tsx
  const { id: teacherId } = authenticatedUser ?? {};
  const ... = useQuery(..., {
    ...
    variables: { teacherId },
    skip: !teacherId,
  });

  // later
  if (!teacherId) {
    return <Redirect to="/login" />;
  }
```

Your way is fine too, it just means that people might run into an error state more frequently.